plugins/gpu/nvidia_gpu_by_user

```diff
@@ -0,0 +1,240 @@
+#!/bin/sh
+#  -*- sh -*-
+
+: <<EOF #
+ =cut
+
+=head1 NAME
+
+gpubyuser - Plugin to monitor GPU memory usage by user
+
+=head1 CONFIGURATION
+
+Add this to node configuration file:
+ [nvidia_gpu_by_user]
+  env.smiexec - Location of nvidia-smi executable.
+
+=head1 AUTHOR
+
+Hideki Takano
+ymhtakano@gmail.com
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+EOF
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+# Get location of nvidia-smi executable or use default
+nvSmiExec=${smiexec:-'/usr/bin/nvidia-smi'}
+
+# Check if autoconf was requested
+if [ "$1" = "autoconf" ]; then
+	# Autoconf only returns yes if nvidia-smi exists and is executable
+	if [ -x "$nvSmiExec" ]; then
+		echo yes
+		exit 0
+	else
+		echo "no (nvidia-smi executable not found)"
+		exit 0
+	fi
+fi
+
+gpuUSERS=$(clean_fieldname "$(ls /home)" | tr "\n" " ")
```

Comment:
This feels slightly dirty to me, since it uses the non-documented behaviour of `clean_fieldname` with regard to multi-line input. Please use something like the following instead:
```shell
find /home -maxdepth 1 -mindepth 1 -type d | while read -r dname; do clean_fieldname "$(basename "$dname")"; done
```

Btw.: is the underlying assumption (relevant users having a home directory below `/home/` that matches their name) a good choice?
How about the following - in order to get at least around the directory/username match requirement?
```shell
getent passwd | cut -d : -f 1,6 | grep ":/home/" | cut -d : -f 1
```
(I cannot judge its portability - maybe you can comment)

Reply:
In some environments "getent passwd" will not return users in LDAP (or even YP/NIS):
```
$ getent passwd | grep niclangf
$ getent passwd niclangf
niclangf:*:7181:10001:niclangf:/home/niclangf:/bin/bash
```
and in some environments "getent passwd" will return many many thousands of users.

What you find in /home is also very dodgy. Home directories can be mounted many other places in large environments.

So getting a "full catalog" of users is not very reliable or scalable.  But if you have either a username or a UID this works

```
$ getent passwd niclangf
niclangf:*:7181:10001:niclangf:/home/niclangf:/bin/bash
$ getent passwd 7181
niclangf:*:7181:10001:niclangf:/home/niclangf:/bin/bash
```

If you want to provide a stable list of users to keep the graphs stable then persist the usernames... but I see I never got around to writing persistence functions for plugin.sh... There are good hints in Munin/Plugin.pm about what environment variables and such to use if you feel wizardly and want to write a file to persist something at least.

Reply:
So what could be a good and portable approach for requesting "all non-system users" (I think, this is the goal)?
I have the feeling, that we cannot provide a really good and platform neutral implementation with the little effort that is justified for this task in this context. Thus I would be happy with anything that does not feel *too* dirty (maybe even the original proposal).

Reply:
There isn't.  Just graph for those that have been using the GPU and somehow persist it.

Reply:
Thank you for good comments.
If there isn't suitable implementation for listing all non-system users,
then I think it is still good for users to be configurable.